contrib/clojure-package/src/org/apache/clojure_mxnet/infer.clj

```diff
@@ -0,0 +1,210 @@
+  WrappedObjectDetector
+  (detect-objects [this image]
+    (detect-objects this image nil))
+  (detect-objects [this image topk]
+    (util/coerce-return-recursive
+     (.imageObjectDetect (:object-detector this)
+                         image
+                         (util/->int-option topk))))
+  (detect-objects-batch [this images]
+    (detect-objects-batch this images nil))
+  (detect-objects-batch [this images topk]
+    (util/coerce-return-recursive
+     (.imageBatchObjectDetect (:object-detector this)
+                              images
+                              (util/->int-option topk))))
+  (detect-objects-with-ndarrays [this input-arrays]
+    (detect-objects-with-ndarrays this input-arrays nil))
+  (detect-objects-with-ndarrays [this input-arrays topk]
+    (util/coerce-return-recursive
+     (.objectDetectWithNDArray (:object-detector this)
+                               (util/vec->indexed-seq input-arrays)
+                               (util/->int-option topk)))))
+
+(defprotocol AInferenceFactory
+  (create-predictor [this] [this opts])
+  (create-classifier [this] [this opts])
+  (create-image-classifier [this] [this opts])
+  (create-object-detector [this] [this opts]))
+
+(defrecord InferenceFactory [model-path-prefix input-descriptors]
+  AInferenceFactory
+  (create-predictor [this]
+    (create-predictor this {}))
+  (create-predictor [this opts]
+    (let [{:keys [contexts epoch]
+           :or {contexts [(context/cpu)] epoch 0}} opts]
+      (->WrappedPredictor
+       (new Predictor
+            model-path-prefix
+            (util/vec->indexed-seq input-descriptors)
+            (into-array contexts)
+            (util/->int-option epoch)))))
+  (create-classifier [this]
+    (create-classifier this {}))
+  (create-classifier [this opts]
+    (let [{:keys [contexts epoch]
+           :or {contexts [(context/cpu)] epoch 0}} opts]
+      (->WrappedClassifier
+       (new Classifier
+            model-path-prefix
+            (util/vec->indexed-seq input-descriptors)
+            (into-array contexts)
+            (util/->int-option epoch)))))
+  (create-image-classifier [this]
+    (create-image-classifier this {}))
+  (create-image-classifier [this opts]
+    (let [{:keys [contexts epoch]
+           :or {contexts [(context/cpu)] epoch 0}} opts]
+      (->WrappedImageClassifier
+       (new ImageClassifier
+            model-path-prefix
+            (util/vec->indexed-seq input-descriptors)
+            (into-array contexts)
+            (util/->int-option epoch)))))
+  (create-object-detector [this]
+    (create-object-detector this {}))
+  (create-object-detector [this opts]
+    (let [{:keys [contexts epoch]
+           :or {contexts [(context/cpu)] epoch 0}} opts]
+      (->WrappedObjectDetector
+       (new ObjectDetector
+            model-path-prefix
+            (util/vec->indexed-seq input-descriptors)
+            (into-array contexts)
+            (util/->int-option epoch))))))
+
```

Comment:
These are great. If you added specs to these main functions it would be beautiful 😍

Reply:
done